src/components/admin/Login.jsx

```diff
@@ -1,73 +1,114 @@
+      // GoogleAPILoaded is a global var, set to true when Google platform.js script loaded
+      if (window.THE_GAZELLE.googleAPILoaded) { // eslint-disable-line no-undef
+        this.renderButton();
+        window.gapi.load('auth2', () => {
+          this.safeSetState({
+            GoogleAPIReady: true,
+          });
 
-  handleSubmit(e) {
-    e.preventDefault();
-    const pass = e.target.password.value;
-    // eslint-disable-next-line max-len
-    if (hash(pass) !== 'eaafc81d7868e1c203ecc90f387acfa4c24d1027134b0bfda6fd7c536efc5d8dd5718609a407dbfcd41e747aec331153d47733153afb7c125c558acba3fb6bcd') {
-      window.alert('Incorrect password');
-      e.target.password.value = ''; // eslint-disable-line no-param-reassign
-    } else {
-      // Otherwise it is correct and we let the site redirect the client
-      let url = this.props.location.query.url || '';
-      if (url) {
-        _.forEach(this.props.location.query, (q, name) => {
-          if (name !== 'url') {
-            url = `${url}&${name}=${q}`;
+          const auth = window.gapi.auth2.getAuthInstance();
+          if (!auth || !auth.isSignedIn.get()) {
+            window.gapi.auth2.init({
+              client_id: googleClientID,
+            });
+          } else { // detects user has signed in
+            this.onSignInSuccess(auth.currentUser.get());
           }
         });
+        clearInterval(interval);
       }
-      browserHistory.push(url);
+    }, 100);
+  }
+
+  handleRedirect() {
+    // Add additional query params to new url
+    let url = this.props.location.query.url || '';
+    if (url) {
+      _.forEach(this.props.location.query, (q, name) => {
+        if (name !== 'url') {
+          url = `${url}&${name}=${q}`;
+        }
+      });
     }
+    browserHistory.push(url);
+  }
+
+  handleLoginCI() {
+    // if CI, just let it login without interacting with Google Login popup window
+    if (isCI) {
+      this.handleRedirect();
+    }
+  }
+
+  onSignInSuccess(response) {
+    const token = { data: response.getAuthResponse().id_token };
+
+    const xhr = new XMLHttpRequest();
+    xhr.open('POST', '/googlelogin');
+    xhr.setRequestHeader('Content-Type', 'application/json');
+
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
+        this.handleRedirect();
+      } else if (xhr.readyState === XMLHttpRequest.DONE) {
+        alert(`Google Login Failed.\n${xhr.status}: ${xhr.statusText}`);
+      }
+    };
+
+    xhr.send(JSON.stringify(token));
+  }
+
+  onSignInFailure(response) {
+    alert(`Google Login Failed.\n${response.error}`);
+  }
+
+  renderButton() {
+    window.gapi.signin2.render('my-signin2', {
+      scope: 'profile email',
+      longtitle: true,
+      onsuccess: this.onSignInSuccess,
+      onfailure: this.onSignInFailure,
+    });
   }
 
   render() {
+    const fade = this.state.GoogleAPIReady ? 1 : 0.5;
     return (
-      <div id="login-page">
```

Comment:
This is also a small thing, but I think there are some nice semantics about leaving this wrapper div with id `login-page`. And if you change this then also change back the diff in the header test that you changed